src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,150 @@
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EventCallback.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Add a generic <see cref="EventCallback"/> parameter with a value for this builder.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector</param>
+        /// <param name="callback">The event callback.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder&lt;TComponent&gt;"/> which can be chained.</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EventCallback<EventArgs>>> parameterSelector, Func<EventArgs, Task> callback)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EventCallback.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Create a <see cref="ComponentParameter"/> collection.
+        /// </summary>
+        /// <returns>A IReadOnlyCollection of <see cref="ComponentParameter"/></returns>
+        public IReadOnlyCollection<ComponentParameter> Build()
+        {
+            return _componentParameters;
+        }
+
+        private static (string name, bool isCascading) GetDetailsFromExpression<T>(Expression<Func<TComponent, T>> parameterSelector)
+        {
+            if (parameterSelector == null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (parameterSelector.Body is MemberExpression memberExpression && memberExpression.Member is PropertyInfo propertyInfo)
+            {
+                var parameterAttribute = propertyInfo.GetCustomAttribute<ParameterAttribute>();
+                var cascadingParameterAttribute = propertyInfo.GetCustomAttribute<CascadingParameterAttribute>();
+                if (parameterAttribute is null && cascadingParameterAttribute is null)
+                {
+                    throw new ArgumentException($"The property '{propertyInfo.Name}' selected by the provided '{parameterSelector}' does not have the [Parameter] or [CascadingParameter] attribute in the component '{typeof(TComponent)}'.");
+                }
+
+                if (parameterAttribute is { })
+                {
+                    return (propertyInfo.Name, false);
+                }
+
+                if (!string.IsNullOrEmpty(cascadingParameterAttribute.Name))
+                {
+                    return (cascadingParameterAttribute.Name, true);
+                }

```

Comment:
```suggestion
                if (!string.IsNullOrEmpty(cascadingParameterAttribute.Name))
                {
                    return (cascadingParameterAttribute.Name, true);
                }
                if(cascadingParameterAttribute is {})
                {
                    return (null, true);                    
                }
```
I _think_ this captures the case where it is an unnamed cascadingvalue.

Reply:
I think this is not needed / not valid / not possible.
See the method "GetDetailsFromExpression" and read the comments...

Reply:
You are right, it should be ok.